### PR TITLE
[Errors](ft): Allowing custom descriptions

### PR DESCRIPTION
**Pull request by alexandre-merle**
- Allow custom descriptions in arsenal errors

Fixes https://github.com/scality/Arsenal/issues/117

### CODE REVIEW

File: tests/unit/errors.js [hunk 1]
```diff
@@ -3,19 +3,23 @@ const assert = require('assert');
...
+                errorsJSON[index].description, 'Incorrect description');
+            assert.strictEqual(errors[index][index], true,
+                `${index} key must be true`);
             done();
         });
     });
+
+    it('should allow custom error descriptions', () => {
+        const error = errors.NoSuchEntity.generateCustom('custom-description');
+        assert.strictEqual(error.description, 'custom-description');
```
Comment (rahulreddy) [inline tests/unit/errors.js]: Let's add another assertion which checks `errors.NoSuchEnitity`'s description is unchanged.

File: lib/errors.js [hunk 2]
```diff
@@ -7,6 +7,10 @@ class ArsenalError extends Error {
         this.description = desc;
         this[type] = true;
     }
+
+    generateCustom(description) {
```
Comment (LaurenSpiegel) [inline lib/errors.js]: let's call this customizeDescription

**Comment by LaurenSpiegel** [comment]
Should we have some customized description options that go with certain policies?  Maybe a long term goal once we see which customized descriptions we are using with which error codes.

**Comment by alexandre-merle** [comment]
Updated :) @LaurenSpiegel yes, could be great to have such feature :)

**Comment by LaurenSpiegel** [comment]
👍

**Comment by rahulreddy** [comment]
This is awesome.
👍

**Comment by alexandre-merle** [comment]
Rebased :)

**Pull request merged**